app/scripts/lib/rpc-method-middleware/handlers/add-ethereum-chain.js

```diff
@@ -294,6 +298,9 @@ async function addEthereumChainHandler(
         fromNetworkConfiguration: getProviderConfig(),
       },
     });
+    if (hasPermission(req.origin)) {
+      setNetworkClientIdForDomain(req.origin, networkConfigurationId);
```

Comment:
This solution seems OK, but it makes me wonder if similar cases exist elsewhere.

Have we considered listening to permission state changes, and automatically setting a selected chain when a new origin is granted permissions? That would ensure that we never "forget" to set this initial selected network client.

Reply:
> Have we considered listening to permission state changes, and automatically setting a selected chain when a new origin is granted permissions

Yes indeed. In fact we're doing that in this PR. See [here](https://github.com/MetaMask/metamask-extension/pull/23130/files#r1503052895).

The problem here is not the **initial** network being set, its that we need to change the networkClientId for a dapp that already is permissioned.

Reply:
Ahh I see,  I follow now, thanks